api/cpp/platform.rs

```diff
@@ -0,0 +1,192 @@
+            );
+            uninit.assume_init()
+        }
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn slint_platform_register(
+    user_data: PlatformUserData,
+    drop: unsafe extern "C" fn(PlatformUserData),
+    window_factory: unsafe extern "C" fn(PlatformUserData, *mut WindowAdapterRcOpaque),
+) {
+    let p = CppPlatform { user_data, drop, window_factory };
+    i_slint_core::platform::set_platform(Box::new(p)).unwrap();
+}
+
+type SoftwareRendererOpaque = *const c_void;
+
+#[no_mangle]
+pub unsafe extern "C" fn slint_software_renderer_new(
+    buffer_age: u32,
+    window: &WindowAdapterRcOpaque,
+) -> SoftwareRendererOpaque {
+    let window = core::mem::transmute::<&WindowAdapterRcOpaque, &Rc<dyn WindowAdapter>>(window);
+    let weak = Rc::downgrade(window);
+    match buffer_age {
+        0 => Box::into_raw(Box::new(SoftwareRenderer::<0>::new(weak))) as SoftwareRendererOpaque,
+        1 => Box::into_raw(Box::new(SoftwareRenderer::<1>::new(weak))) as SoftwareRendererOpaque,
+        2 => Box::into_raw(Box::new(SoftwareRenderer::<2>::new(weak))) as SoftwareRendererOpaque,
+        _ => unreachable!(),
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn slint_software_renderer_drop(buffer_age: u32, r: SoftwareRendererOpaque) {
+    match buffer_age {
+        0 => drop(Box::from_raw(r as *mut SoftwareRenderer<0>)),
+        1 => drop(Box::from_raw(r as *mut SoftwareRenderer<1>)),
+        2 => drop(Box::from_raw(r as *mut SoftwareRenderer<2>)),
+        _ => unreachable!(),
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn slint_software_renderer_render_rgb8(
+    buffer_age: u32,
+    r: SoftwareRendererOpaque,
+    buffer: *mut Rgb8Pixel,
+    buffer_len: usize,
+    buffer_stride: usize,
+) {
+    let buffer = core::slice::from_raw_parts_mut(buffer, buffer_len);
+    match buffer_age {
+        0 => (*(r as *const SoftwareRenderer<0>)).render(buffer, buffer_stride),
+        1 => (*(r as *const SoftwareRenderer<1>)).render(buffer, buffer_stride),
+        2 => (*(r as *const SoftwareRenderer<2>)).render(buffer, buffer_stride),
+        _ => unreachable!(),
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn slint_software_renderer_handle(
+    buffer_age: u32,
+    r: SoftwareRendererOpaque,
+) -> RendererPtr {
+    let r = match buffer_age {
+        0 => (r as *const SoftwareRenderer<0>) as *const dyn Renderer,
+        1 => (r as *const SoftwareRenderer<1>) as *const dyn Renderer,
+        2 => (r as *const SoftwareRenderer<2>) as *const dyn Renderer,
+        _ => unreachable!(),
+    };
+    core::mem::transmute(r)
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn slint_windowrc_has_active_animations(
+    handle: *const WindowAdapterRcOpaque,
+) -> bool {
+    let window_adapter = &*(handle as *const Rc<dyn WindowAdapter>);
+    window_adapter.window().has_active_animations()
+}
+
+#[no_mangle]
+pub extern "C" fn slint_platform_update_timers_and_animations() {
```

Comment:
I suppose the function for the next expiring timer will be needed, too?

Reply:
Yes, true. But so far there is no timer in the poc